src/inventory.cpp

```diff
@@ -521,95 +545,33 @@ void inventory::form_from_map( map &m, std::vector<tripoint> pts, const Characte
             }
         }
 
-        // WARNING: The part below has a bug that's currently quite minor
-        // When a vehicle has multiple faucets in range, available water is
-        //  multiplied by the number of faucets.
-        // Same thing happens for all other tools and resources, but not cargo
         const optional_vpart_position vp = m.veh_at( p );
         if( !vp ) {
             continue;
         }
-        vehicle *const veh = &vp->vehicle();
-
-        //Adds faucet to kitchen stuff; may be horribly wrong to do such....
-        //ShouldBreak into own variable
-        const cata::optional<vpart_reference> kpart = vp.part_with_feature( "KITCHEN", true );
-        const cata::optional<vpart_reference> faupart = vp.part_with_feature( "FAUCET", true );
-        const cata::optional<vpart_reference> weldpart = vp.part_with_feature( "WELDRIG", true );
-        const cata::optional<vpart_reference> craftpart = vp.part_with_feature( "CRAFTRIG", true );
-        const cata::optional<vpart_reference> forgepart = vp.part_with_feature( "FORGE", true );
-        const cata::optional<vpart_reference> kilnpart = vp.part_with_feature( "KILN", true );
-        const cata::optional<vpart_reference> chempart = vp.part_with_feature( "CHEMLAB", true );
-        const cata::optional<vpart_reference> cargo = vp.part_with_feature( "CARGO", true );
-
-        if( cargo ) {
-            const vehicle_stack items = veh->get_items( cargo->part_index() );
+        const vehicle &veh = vp->vehicle();
```

Comment:
all of the following also needs to go into vehicle_use.cpp.  The code reduction is nice but please couple it with a code reorganizaion.

Reply:
This should now be ready in these 3 commits https://github.com/CleverRaven/Cataclysm-DDA/pull/44700/files/7fd8438114162b6f1de4e3e8d5d1b2515db772f7..f0804737ffd7f91d63af2d6e1d4f9c9f7155b722

I moved provide_pseudo_item and the backing std::set member<itype_id> for it to inventory and put a section in the docs